### PR TITLE
initial support for upload post-processor

### PR DESCRIPTION
**Pull request by vtolstov**
Signed-off-by: Vasiliy Tolstov <v.tolstov@selfip.ru>

### CODE REVIEW

**Comment by samdunne** [comment]
+1

**Comment by imduffy15** [comment]
+1 any reason for not merging this? Is there work left to be done?

@cbednarski whats the release date like for "upcoming milestone"

**Comment by cbednarski** [comment]
Thanks for the PR! I think this needs a few things before it's ready to merge. At minimum, we'll need tests and docs. Can you add a basic config file that shows how to use this?

If this is intended just for glance, rather than a generic upload post-processor, I think the name / docs should indicate that.

File: post-processor/upload/post-processor.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+				}
+			case "keyfile":
+				upl.KeyFile = p[1]
+			}
+		}
+	}
+	return upl, nil
+}
+
+func (u *UploaderSSH) Upload(artifact packer.Artifact) error {
```
Comment (cbednarski) [inline post-processor/upload/post-processor.go]: This function pretty similar to some of the file provisioner code; is there a way we can share the transport logic so it doesn't need to be duplicated?

File: post-processor/upload/post-processor.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+type UploaderSSH struct {
+	Host    string
+	Port    string
+	Login   string
+	Passw   string
+	KeyFile string
+	Path    string
+	File    string
+	Timeout time.Duration
```
Comment (cbednarski) [inline post-processor/upload/post-processor.go]: Can we use an SSH communicator struct here so we can keep the same functionality, validation, and configuration scheme as other packer plugins?

**Comment by cbednarski** [comment]
@imduffy15 If / when I have concrete plans for the next release I try to include it in the milestone. I don't have that info right now.

**Comment by imduffy15** [comment]
Awesome! Thanks @cbednarski

**Comment by vtolstov** [comment]
Thanks for suggestions, i can reuse ssh communicator code, but i don't want to use it variables in post-processor config because my proposed DSN much simpler and universal.

**Comment by vtolstov** [comment]
close now, because we have shell-local post processor and i want to complete upload post-processor before..

**Pull request closed without merging**